### PR TITLE
Issue #SB-22936 chore: Program List API refactoring

### PR DESCRIPTION
**Pull request by ahghatol**

### CODE REVIEW

File: src/helpers/programHelper.js [hunk 1]
```diff
@@ -920,6 +923,140 @@ class ProgramServiceHelper {
...
+    const nomination = data.request.filters.nomination;
+    const user_id = _.get(nomination, 'user_id');
+    const organisation_id = _.get(nomination, 'organisation_id');
+    const status = _.get(nomination, 'status');
+    const roles = _.get(nomination, 'roles');
+
+    // Contributor org user (my projects)
+    if(user_id && organisation_id && status && roles) {
+      const res = await this.getContribUserPrograms(data, filters);
+      let aggregatedRes = [];
```
Comment (venkateshwarans) [inline src/helpers/programHelper.js:937]: Do this in a`try catch`

File: src/service/programService.js [hunk 2]
```diff
@@ -1228,51 +1131,112 @@ function programList(req, response) {
...
           apiId: 'api.program.list',
           ver: '1.0',
           msgid: uuid(),
           responseCode: 'ERR_LIST_PROGRAM',
           result: err
-        },errCode+errorCodes.CODE5));
+        },errCode+errorCodes.CODE4));
       });
+    } else {
+      model.program.findAll({
```
Comment (venkateshwarans) [inline src/service/programService.js:1196]: let us use `await` and `try catch` to handle promises. we won't end up in nested functions.

**Review by venkateshwarans — COMMENTED** [review]
check comments

File: src/service/programService.js (hunk 2 above)
Comment (ahghatol) [reply src/service/programService.js]: Fixed

File: src/helpers/programHelper.js (hunk 1 above)
Comment (ahghatol) [reply src/helpers/programHelper.js]: Added try-catch in the parent function. If error any error then the parent function will catch it and throw an error in API resp.

**Pull request merged**